logstash-core-event-java/src/main/java/com/logstash/RubyJavaObject.java

```diff
@@ -0,0 +1,124 @@
+package com.logstash;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import org.jruby.Ruby;
+import org.jruby.RubyNil;
+import org.jruby.runtime.builtin.IRubyObject;
+
+import java.io.InvalidObjectException;
+import java.io.ObjectStreamException;
+import java.io.Serializable;
+import java.io.ObjectInputStream;
+
+public final class RubyJavaObject implements Serializable {
+    private static final long serialVersionUID = 4358368393114318978L;
+
+    private transient IRubyObject rubyValue;
+    private transient boolean isRubyNil;
+    private Object javaValue;
+
+    public RubyJavaObject(Object javaValue) {
```

Comment:
the fact that we did not create a class for holding Java values and have opted for `Object` result in having method overloading relying on the implicit more specific type rule to pick the right method which is not very clear IMO. It is probably too big a change to consider refactoring this to use a "proper" object model for values, but maybe we could a t least add a few unit tests to show that the `RubyJavaObject` method overloading really work?